packages/mantine/src/components/table/table-edit-columns-visibility/TableEditColumnsVisibility.tsx

```diff
@@ -0,0 +1,90 @@
+import {Button, Checkbox, Divider, Grid, Popover, ScrollArea, Stack} from '@mantine/core';
+import {FunctionComponent} from 'react';
+
+import {TableComponentsOrder} from '../Table.styles';
+import {useTable} from '../TableContext';
+import useStyles from './TableEditColumnsVisibility.styles';
+import {TableEditColumnsVisibilityProps} from './TableEditColumnsVisibility.types';
+
+const COLUMNS_IDS_TO_EXCLUDE = ['collapsible', 'select'];
+
+export const TableEditColumnsVisibility: FunctionComponent<TableEditColumnsVisibilityProps> = ({
+    classNames,
+    styles,
+    unstyled,
+    label = 'Edit columns',
+    showVisibleCountLabel = false,
+    nonHideableColumns,
+    maxSelectableColumns,
+    columnNames,
+}) => {
+    const {classes} = useStyles(null, {name: 'TableEditColumnsVisibility', classNames, styles, unstyled});
+    const {getAllColumns} = useTable();
+
+    const filteredColumns = getAllColumns().filter((column) => !COLUMNS_IDS_TO_EXCLUDE.includes(column.id));
+
+    const selectedColumnsCount = filteredColumns.filter((column) => column.getIsVisible()).length;
+    const adjustedMaxSelectableColumns = maxSelectableColumns
+        ? Math.min(maxSelectableColumns, filteredColumns.length)
+        : filteredColumns.length;
+
+    const remainingColumnsCount = adjustedMaxSelectableColumns
+        ? adjustedMaxSelectableColumns - selectedColumnsCount
+        : filteredColumns.filter((column) => !column.getIsVisible() && !nonHideableColumns?.includes(column.id)).length;
+
+    let footerText;
+    if (selectedColumnsCount === filteredColumns.length) {
+        footerText = 'All available columns selected';
+    } else if (adjustedMaxSelectableColumns && selectedColumnsCount >= adjustedMaxSelectableColumns) {
+        footerText = 'Maximum columns selected';
+    } else {
+        footerText = `You can choose ${remainingColumnsCount} more`;
+    }
+
+    if (filteredColumns.length <= 0) {
+        return null;
+    }
+
+    return (
+        <Grid.Col span="content" order={TableComponentsOrder.EditColumnsVisibility} py="sm" className={classes.root}>
+            <Popover position="bottom" shadow="md">
+                <Popover.Target>
+                    <Button variant="outline">{`${label}${
```

Comment:
we might need to pass the button variant down as a prop too, since i think the designs for the commerce-hub are using the `secondary` variant rather than `outline`, but `outline` probably doesn't exist in the plasma theme

Reply:
outline is a default variant ine mantine and is used as is IIRC 🤔 But I agree with your point anyway, better safe than sorry 👍